### ISSUE TITLE
SPIKE: How best to enable front end to display/calculate next event

### ISSUE THREAD

**Issue by jtmst** [department-of-veterans-affairs/va.gov-cms#9979]
## Description
Unblocks #9576

Add the logic to determine "next event time" for all events as a computed field.  Make that new data available to the front end.
**UPDATE** – Per comments, consider FE solutions as well, as BE solutions can be correct at content build but go stale (feature past events) as time passes between content releases. 

## Acceptance Criteria
- [ ] Next event time is determined
- [ ] Next event time is available to the front end

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by jilladams** [comment]
Per Slack chat: 

<img width="770" alt="Screen Shot 2022-07-29 at 4 00 51 PM" src="https://user-images.githubusercontent.com/85581471/181836459-4b358246-9f13-406a-a413-f33ddf7c313f.png">

Need this SPIKE to understand if the proposed computed field makes sense from backend perspective. If so: will execute on it, and if not: work goes back to FE to solve. We think.

**Comment by swirtSJW** [comment]
@jtmst and @dsasser and @ryguyk  I don't think we can handle this as a computed field in Drupal... or at least it is not ideal.   The decoupled nature of our site creates a situation where this kind of thing would happen:

- The last content release of the week is Friday at 6pm  So the computed field might reveal that a friday night event is the most upcoming. 
- Saturday and Sunday would feature an event that has already passed.

Ethan built FE a function that when called for an event gives the most upcoming instance of the event.  I think we want to still leverage that and use data that is already on the FE..The real problem is the data location.  That payload of events data is only on the event listing page.  We actually need it in a place that both pages could access it.  Then the backfill of "Featured" could happen in real time.   When one expired, the next one would take its place regardless of whether there was a content release or not.

 I'd prefer to put time into solving the actual problem and not just band-aid it with a new computed field from the CMS that perpetuates one of the problems.

I have some ideas for pulling this off, but have not connected all the pieces of the puzzle.  We should kick this around a bit and come up with a solution that solves this problem in a more data centric way.

**Comment by dsasser** [comment]
@swirtSJW thanks for the feedback on this spike. This was intended to serve as a research ticket to unpack if a BE solution was viable--and you hit on one of the issues I had considered after making the original suggestion about a computed field. Looking forward to hearing about your suggested approaches.

**Comment by ryguyk** [comment]
The comment above by @swirtSJW captures very well why a back-end solution is likely a non-starter. The solution should be handled by the front end because it should be calculated based on the time the page is loaded.  It'll require two things:

1) Ensure sufficient event data is making it to the front-end page in question. If not, update GraphQL.
2) Write logic to calculate the next upcoming event from a list of events.

This ticket should be closed in favor of implementing the changes under #9576.

Note:

When we met about this, there was also some larger discussion around the potential of automating the generation of data files for events at VAMC Systems (and possibly other content types).  I've created [a new ticket to capture that](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10003).
